### ISSUE TITLE
Crash on SIGINT

### ISSUE THREAD

**Issue by ePirat** [OpenVisualCloud/SVT-AV1#1172]
When terminating an encode with Ctrl-C (SIGINT) in a debug build with address sanitizer it crashes:

```
$ ./Bin/Release/SvtAv1EncApp -asm 0 -i ~/fireworks_3840x2160.yuv -n 450 -w 3840 -h 2160 -b ~/out.ivf                                                                 130 ↵
-------------------------------------------
SVT-AV1 Encoder
SVT [version]:	SVT-AV1 Encoder Lib v0.8.1
SVT [build]  :	GCC 4.2.1 Compatible Apple LLVM 11.0.3 (clang-1103.0.32.29)	 64 bit
LIB Build date: Mar 30 2020 14:01:35
-------------------------------------------
Number of logical cores available: 4
Number of PPCS 72
[asm level on system : up to avx2]
[asm level selected : up to c]
------------------------------------------- 
SVT [config]: Main Profile	Tier (auto)	Level (auto)	
SVT [config]: EncoderMode 							: 8 
SVT [config]: EncoderBitDepth / EncoderColorFormat / CompressedTenBitFormat	: 8 / 1 / 0
SVT [config]: SourceWidth / SourceHeight					: 3840 / 2160 
SVT [config]: FrameRate / Gop Size						: 30 / 32 
SVT [config]: HierarchicalLevels  / PredStructure		: 4 / 2
SVT [config]: BRC Mode / QP  / LookaheadDistance / SceneChange			: CQP / 50 / 33 / 0 
------------------------------------------- 
warning: -asm will be deprecated soon, please use --asm
Encoding          ^CAddressSanitizer:DEADLYSIGNAL
=================================================================
==4237==ERROR: AddressSanitizer: SEGV on unknown address 0x000000000028 (pc 0x00010a9877a2 bp 0x7ffee54adc30 sp 0x7ffee54adc00 T0)
==4237==The signal is caused by a READ memory access.
==4237==Hint: address points to the zero page.
    #0 0x10a9877a1 in eb_get_empty_object EbSystemResourceManager.c:598
    #1 0x10ae86d63 in eb_svt_enc_send_picture EbEncHandle.c:3472
    #2 0x10a76bff6 in process_input_buffer EbAppProcessCmd.c:952
    #3 0x10a767230 in main EbAppMain.c:212
    #4 0x7fff72ebe7fc in start (libdyld.dylib:x86_64+0x1a7fc)

==4237==Register values:
rax = 0x0000000000000028  rbx = 0x0000000000000000  rcx = 0x0000100000000005  rdx = 0x00001fffdca95b8c  
rdi = 0x0000606000d7d9f8  rsi = 0x0000000000000000  rbp = 0x00007ffee54adc30  rsp = 0x00007ffee54adc00  
 r8 = 0x00007ffee54adc60   r9 = 0xffffffff00000000  r10 = 0x0000606000d7da58  r11 = 0x0000000000000246  
r12 = 0x0000100000000000  r13 = 0x0000606000d7d9e0  r14 = 0x0000606000d7d9f0  r15 = 0x00001c0c001afb3e  
AddressSanitizer can not provide additional info.
SUMMARY: AddressSanitizer: SEGV EbSystemResourceManager.c:598 in eb_get_empty_object
==4237==ABORTING
[1]    4237 abort      ./Bin/Release/SvtAv1EncApp -asm 0 -i ~/fireworks_3840x2160.yuv -n 450 -w 3840
```

**Comment by mazharaIntel** [comment]
Attempted to reproduce the issue in windows and linux using debug and release mode for the svt-av1 encoder. The issue could not be reproduced.

**Comment by ePirat** [comment]
Did you build with Address sanitizer? Additionally, cancel the encoder before it actually encoded any frames.
I can try again to reproduce tomorrow, please let me know what additional info you need that I should provide you when I manage to reproduce.

**Comment by mazharaIntel** [comment]
I have not tried address sanitizer yet. Working on it. Were you able to reproduce the issue?

**Comment by mazharaIntel** [comment]
Tried to reproduce with address sanitizer as well. Still could not reproduce it for debug and release build.

**Comment by ePirat** [comment]
You have to cancel it while it between the allocation stats output and it actually finishing encoding the first frame. I did manage to reproduce it again today when I tested.

**Comment by mazharaIntel** [comment]
I have been stopping the encoder with ctrl+c between the allocation stats output and before it finishes the first frame. Is it reproducible for different resolution and clips?

**Comment by ePirat** [comment]
A specific clip, I currently do not have any other to test around. I've linked the specific file in a previous issue #959

**Comment by hassount** [comment]
still happening ?

**Comment by ePirat** [comment]
Can't reproduce on macOS anymore, probably due to change to dispatch semaphores.
On linux this will likely be fixed by #1358
